### PR TITLE
Only consult applocal userconfig if metrics are enabled

### PR DESCRIPTION
**Pull request by ras0219-msft**
If metrics are not enabled, we do not need to read and write the user config file.

### CODE REVIEW

**Review by vicroms — APPROVED** [review]

File: src/vcpkg.cpp [hunk 1]
```diff
@@ -259,6 +260,11 @@ int main(const int argc, const char* const* const argv)
         }
     } // unlock g_metrics
 
+    if (metrics_enabled)
+    {
+        load_config(fs);
```
Comment (BillyONeal) [inline src/vcpkg.cpp:265]: This thing being called "load_config" and having lots of "spooky action at a distance" seems like a time bomb that is likely to regress.

How about changing the design such that metrics are off, then having metrics->enable_metrics() which has the load_config() guts inside?

**Review by BillyONeal — COMMENTED** [review]
Is this something you can write an (e2e or otherwise) test for?

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
I am happy that my comments were addressed here and think this is ready to land as soon as builds and tests pass.

**Comment by ras0219-msft** [comment]
Many recursive locking bugs later, success!

**Pull request merged**